osu.Game/Online/Multiplayer/MultiplayerClient.cs

```diff
@@ -446,7 +446,25 @@ Task IMultiplayerClient.HostChanged(int userId)
 
         Task IMultiplayerClient.SettingsChanged(MultiplayerRoomSettings newSettings)
         {
-            Scheduler.Add(() => updateLocalRoomSettings(newSettings));
+            Debug.Assert(APIRoom != null);
+            Debug.Assert(Room != null);
+
+            Scheduler.Add(() =>
+            {
+                // ensure the new selected item is populated immediately.
+                var playlistItem = APIRoom.Playlist.SingleOrDefault(p => p.ID == newSettings.PlaylistItemId);
+
+                if (playlistItem != null)
+                {
+                    GetAPIBeatmap(playlistItem.BeatmapID).ContinueWith(b =>
+                    {
+                        Scheduler.Add(() => playlistItem.Beatmap.Value = b.Result);
+                    }, TaskContinuationOptions.OnlyOnRanToCompletion);
+                }
+
+                updateLocalRoomSettings(newSettings);
```

Comment:
I believe this is going to be a problem after your recent changes, because `RoomSubScreen` doesn't bind to changes to `Beatmap` on the current item. Meaning, inside this schedule, `updateLocalRoomSettings()` will complete first (setting `CurrentMatchPlayingItem`) and only after will `item.Beatmap` be populated via the async + schedule above.

Perhaps you're able to move this inside the nested schedule to fix it.

Reply:
I did consider this but couldn't find that said usage. That's very hidden logic... wonder if we can improve that.

Reply:
Something like this?

```diff
diff --git a/osu.Game/Online/Multiplayer/MultiplayerClient.cs b/osu.Game/Online/Multiplayer/MultiplayerClient.cs
index 82844af3c5..c2b9727309 100644
--- a/osu.Game/Online/Multiplayer/MultiplayerClient.cs
+++ b/osu.Game/Online/Multiplayer/MultiplayerClient.cs
@@ -458,11 +458,17 @@ Task IMultiplayerClient.SettingsChanged(MultiplayerRoomSettings newSettings)
                 {
                     GetAPIBeatmap(playlistItem.BeatmapID).ContinueWith(b =>
                     {
-                        Scheduler.Add(() => playlistItem.Beatmap.Value = b.Result);
-                    }, TaskContinuationOptions.OnlyOnRanToCompletion);
-                }
+                        bool success = b.IsCompletedSuccessfully;
+
+                        Scheduler.Add(() =>
+                        {
+                            if (success)
+                                playlistItem.Beatmap.Value = b.Result;
 
-                updateLocalRoomSettings(newSettings);
+                            updateLocalRoomSettings(newSettings);
+                        });
+                    });
+                }
             });
 
             return Task.CompletedTask;
```

Not sure about nesting the update that deep, but seems to work...